src/DynamoCoreWpf/ViewModels/PackageManager/PackageViewModel.cs

```diff
@@ -410,6 +410,25 @@ private bool IsOwner()
             return packageManagerClient.DoesCurrentUserOwnPackage(Model, dynamoModel.AuthenticationManager.Username);
         }
 
+        private bool CanPublishNewPackage()
+        {
+            if (!CanPublish) return false;
+
+            return packageManagerClient.DoesCurrentUserOwnPackage(Model, dynamoModel.AuthenticationManager.Username) ||
```

Comment:
Does the `DoesCurrentUserOwnPackage` check mean that the package cannot be republished by someone who hasn't published it before?

Reply:
It looks like only an existing package maintainer or package copyright holder can republish (publish a new version) the package? The code would be clearer if you said something like:
```
bool isUserOwnerOrHolder = packageManagerClient.DoesCurrentUserOwnPackage(Model, dynamoModel.AuthenticationManager.Username) || PackageNotPublishedAndUserIsHolder(Model, dynamoModel.AuthenticationManager.Username);
return isUserOwnerOrHolder;
```

Reply:
Wait..Copyright Holder should not be able to publish a version or a package, we do not validate copyright holder on server side, any actions related to package publishing either republishing from local copy or publishing a version should be limited to its current maintainers only.

Reply:
Understood.